deepmd/pt/loss/property.py

```diff
@@ -69,34 +88,63 @@ def forward(self, input_dict, model, label, natoms, learning_rate=0.0, mae=False
         model_pred = model(**input_dict)
-        assert label["property"].shape[-1] == self.task_dim
-        assert model_pred["property"].shape[-1] == self.task_dim
+        nbz = model_pred[self.property_name].shape[0]
+        assert model_pred[self.property_name].shape == (nbz, self.task_dim)
+        assert label[self.property_name].shape == (nbz, self.task_dim)
+        if not self.intensive:
+            model_pred[self.property_name] = model_pred[self.property_name] / natoms
+            label[self.property_name] = label[self.property_name] / natoms
+
+        if self.out_std is None:
+            out_std = model.atomic_model.out_std[0][0]
+        else:
+            out_std = torch.tensor(
+                self.out_std, dtype=env.GLOBAL_PT_FLOAT_PRECISION, device=env.DEVICE
+            )
+        if out_std.shape != (self.task_dim,):
+            raise ValueError(
+                f"Expected out_std to have shape ({self.task_dim},), but got {out_std.shape}"
+            )
+
+        if self.out_bias is None:
+            out_bias = model.atomic_model.out_bias[0][0]
+        else:
+            out_bias = torch.tensor(
+                self.out_bias, dtype=env.GLOBAL_PT_FLOAT_PRECISION, device=env.DEVICE
+            )
+        if out_bias.shape != (self.task_dim,):
+            raise ValueError(
+                f"Expected out_bias to have shape ({self.task_dim},), but got {out_bias.shape}"
+            )
+
         loss = torch.zeros(1, dtype=env.GLOBAL_PT_FLOAT_PRECISION, device=env.DEVICE)[0]
         more_loss = {}
 
         # loss
         if self.loss_func == "smooth_mae":
             loss += F.smooth_l1_loss(
-                label["property"],
-                model_pred["property"],
+                (label[self.property_name] - out_bias) / out_std,
+                (model_pred[self.property_name] - out_bias) / out_std,
                 reduction="sum",
                 beta=self.beta,
             )
         elif self.loss_func == "mae":
             loss += F.l1_loss(
-                label["property"], model_pred["property"], reduction="sum"
+                (label[self.property_name] - out_bias) / out_std,
+                (model_pred[self.property_name] - out_bias) / out_std,
+                reduction="sum",
             )
         elif self.loss_func == "mse":
             loss += F.mse_loss(
-                label["property"],
-                model_pred["property"],
+                (label[self.property_name] - out_bias) / out_std,
+                (model_pred[self.property_name] - out_bias) / out_std,
                 reduction="sum",
             )
         elif self.loss_func == "rmse":
             loss += torch.sqrt(
                 F.mse_loss(
-                    label["property"],
-                    model_pred["property"],
+                    (label[self.property_name] - out_bias) / out_std,
+                    (model_pred[self.property_name] - out_bias) / out_std,
                     reduction="mean",
```

Comment:
Why some reductions are 'sum' and some are 'mean'? It's better to use the average ('mean') over all frames as the loss?

Reply:
![image](https://github.com/user-attachments/assets/718b7ec2-8486-41b9-9977-e84dbfc54d60)
I find that in `ener` loss.
- Reductions are 'mean' when using l2 loss
- Reductions are 'sum' when using l1 loss
I continued above approach.